### PR TITLE
[INLONG-10526][Sort] ClsSink support switch metadata acquire mode

### LINKED ISSUE #10526: [Improve][Sort] ClsSink support switch metadata acquire mode
### Description

_No response_

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
Fixes #10526 

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Comment by XiaoYou201** [comment]
LGTM～

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java [hunk 1]
```diff
@@ -97,20 +101,24 @@ public void reload() {
                 }
             });
 
-            SortTaskConfig newSortTaskConfig = SortConfigHolder.getTaskConfig(taskName);
-            if (newSortTaskConfig == null || newSortTaskConfig.equals(sortTaskConfig)) {
+            TaskConfig newTaskConfig = SortConfigHolder.getTaskConfig(taskName);
+            SortTaskConfig newSortTaskConfig = SortClusterConfigHolder.getTaskConfig(taskName);
+            if ((newTaskConfig == null || newTaskConfig.equals(taskConfig))
```
Comment (luchunliang) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java:106]: The processing logic of TaskConfig and SortTaskConfig can be placed in two implementation classes that implement the same Handler interface, which will be clearer and more convenient for subsequently phasing out the old processing implementation classes.

**Review by luchunliang — DISMISSED** [review]

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java [hunk 2]
```diff
@@ -134,18 +142,41 @@ private void reloadHandler() {
...
-    private void reloadClients() {
+    private Map<String, ClsIdConfig> reloadIdParamsFromTaskConfig(SortTaskConfig sortTaskConfig)
+            throws JsonProcessingException {
+        if (sortTaskConfig == null) {
+            return new HashMap<>();
+        }
+        List<Map<String, String>> idList = this.sortTaskConfig.getIdParams();
+        Map<String, ClsIdConfig> newIdConfigMap = new ConcurrentHashMap<>();
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```
Comment (aloyszhang) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java:167]: `objectMapper` doesn't need created multi times

**Comment by aloyszhang** [comment]
Overall LGTM, just left a comment, PTAL.
And fix the failed unit test also.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java (hunk 2 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java]: fixed, thx

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java (hunk 1 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java:106]: The ClsSinkContext is not an interface or abstract class, the two implementations will result in massive file modification. Same as other sinks

**Review by aloyszhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

**Pull request merged**